File: Low Level Design (LLD)/Problems/06. Splitwise/Java (with Springboot)/splitwise/src/main/java/com/splitwise/splitwise/service/UserService.java
package com.splitwise.splitwise.service;

import com.splitwise.splitwise.model.User;

import java.util.*;

public class UserService {
    private Map<String, User> users = new HashMap<>();

    public User createUser(String userId, String name, String email, String phoneNumber) {
        User user =  new User(userId, name, email, phoneNumber);
        users.put(userId, user);
        return user;
    }

    public User getUser(String userId) {
        return users.get(userId);
    }

    public List<User> getAllUsers() {
        return new ArrayList<>(users.values()); // Assuming 'users' is a Map<String, User>
    }

}

File: Low Level Design (LLD)/Problems/06. Splitwise/Cpp/service/src/UserService.cpp
#include "../include/UserService.h"

std::shared_ptr<User> UserService::createUser(const std::string& userId,
                                              const std::string& name,
                                              const std::string& email,
                                              const std::string& phoneNumber) {
    auto user = std::make_shared<User>(userId, name, email, phoneNumber);
    users[userId] = user;
    return user;
}

std::shared_ptr<User> UserService::getUser(const std::string& userId) const {
    auto it = users.find(userId);
    return it != users.end() ? it->second : nullptr;
}

std::vector<std::shared_ptr<User>> UserService::getAllUsers() const {
    std::vector<std::shared_ptr<User>> allUsers;
    for (const auto& [id, user] : users) {
        allUsers.push_back(user);
    }
    return allUsers;
}
